### ISSUE TITLE
Remove excessive header landmarks, add appropriate one

### ISSUE THREAD

**Issue by jvine** [projectblacklight/arclight#622]
Every document title in the collection list and search results is presented as:

```
<header class="documentHeader row">
  <h3 class="index_title..."
</header>
```

The `<header>` element is meant to contain the site header content that's consistent at the top of every page; ideally each page would have only one `<header>`. 

- [ ] Change all current `<header>` elements to `<div>` elements
- [ ] Add a new `<header>` element that surrounds:
  - top `<nav>` element
  - masthead
  - search bar
